### PR TITLE
Fix dplr error

### PR DESCRIPTION
**Pull request by Yi-FanLi**
DPLR is broken due to the change of coeff(int narg, char **arg) function. The current coeff function requires that the number of types in LAMMPS no more than that in model. In dplr, however, there might be more types in LAMMPS than in the model. This check should be bypassed.

### CODE REVIEW

**Review by njzjz — DISMISSED** [review]
Approve, although I hope a test

**Comment by Yi-FanLi** [comment]
I added a test. However, I am not able to set up the pytest environment on my local computer right now. I pushed the test to this branch, and let's see how github actions perform on this test.

**Comment by njzjz** [comment]
I add `select_real_atoms` to model deviation in https://github.com/Yi-FanLi/deepmd-kit/pull/2. It seems to me that the model deviation never works for DPLR?

**Review by njzjz — APPROVED** [review]
Ok, the tests have passed

File: source/lmp/tests/data_type_map2.lmp
Comment (wanghan-iapcm) [inline source/lmp/tests/data_type_map2.lmp:1]: Please do not commit the data file. write one by the setUp of your test, then remove the file at tearDown.

File: source/lmp/tests/data2.lmp
Comment (wanghan-iapcm) [inline source/lmp/tests/data2.lmp:1]: Please do not commit the data file. write one by the setUp of your test, then remove the file at tearDown.

File: source/api_cc/src/DeepPot.cc [hunk 1]
```diff
@@ -1577,31 +1605,54 @@ void DeepPotModelDevi::compute(
...
+  dcoord.resize(nframes * nall_real * 3);
+  datype.resize(nall_real);
+  // fwd map
+  select_map<VALUETYPE>(dcoord, dcoord_, fwd_map, 3, nframes, nall_real, nall);
+  select_map<int>(datype, datype_, fwd_map, 1);
+  // aparam
+  if (daparam > 0) {
+    aparam.resize(nframes * nloc_real);
+    select_map<VALUETYPE>(aparam, aparam_, fwd_map, daparam, nframes, nloc_real,
+                          nall - nghost);
```
Comment (wanghan-iapcm) [inline source/api_cc/src/DeepPot.cc:1634]: The selection of real atoms looks similar across different `compute` methods. Could you please pack them in one function and call from the `compute`s the function?

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**